src/Sql/Sql/Elastic Jobs/Cmdlet/JobExecution/GetAzureSqlElasticJobExecution.cs

```diff
@@ -0,0 +1,292 @@
+using Microsoft.Azure.Commands.Sql.ElasticJobs.Model;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.ElasticJobs.Cmdlet.JobExecution
+{
+    /// <summary>
+    /// Defines the Get-AzureRmSqlElasticJobExecution Cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlElasticJobExecution",
+        DefaultParameterSetName = DefaultParameterSet)]
+    [OutputType(typeof(AzureSqlElasticJobExecutionModel))]
+    public class GetAzureSqlElasticJobExecution : AzureSqlElasticJobExecutionCmdletBase<AzureSqlElasticJobAgentModel>
+    {
+        /// <summary>
+        /// The parameter sets
+        /// </summary>
+        protected const string GetRootJobExecution = "WithJobExecutionId";
+        protected const string GetRootJobExecutionByParentObject = "WithJobExecutionIdUsingParentObject";
+        protected const string GetRootJobExecutionByParentResourceId = "WithJobExecutionIdUsingParentResourceId";
+
+        /// <summary>
+        /// Gets or sets the resource group name
+        /// </summary>
+        [Parameter(ParameterSetName = DefaultParameterSet,
+            Mandatory = true,
+            Position = 0,
+            HelpMessage = "The resource group name.")]
+        [Parameter(ParameterSetName = GetRootJobExecution,
+            Mandatory = true,
+            Position = 0,
+            HelpMessage = "The resource group name.")]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter]
+        public override string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the agent object input model
+        /// </summary>
+        [Parameter(
+            ParameterSetName = InputObjectParameterSet,
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = "The job execution id.")]
+        [Parameter(
+            ParameterSetName = GetRootJobExecutionByParentObject,
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = "The job execution id.")]
+        [ValidateNotNullOrEmpty]
+        public AzureSqlElasticJobAgentModel ParentObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets the agent resource id
+        /// </summary>
+        [Parameter(
+            ParameterSetName = ResourceIdParameterSet,
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The agent resource id.")]
+        [Parameter(
+            ParameterSetName = GetRootJobExecutionByParentResourceId,
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The agent resource id.")]
+        [ValidateNotNullOrEmpty]
+        public string ParentResourceId { get; set; }
```

Comment:
@johnpaulkee same comment about `-ResourceId`

Reply:
As discussed we'll address the Resource Id changes in a separate PR to reduce complexity of this one. We can do a one-off release with these changes and then add -ResourceId parameter  later.